api/app/Repositories/UserRepository.php

```diff
@@ -51,4 +51,27 @@ public function makeLoginToken($id)
 
         return $token;
     }
+
+    /**
+     * Create or replace an entity by id.
+     *
+     * @param  string  $id
+     * @param  array   $data
+     * @return array
+     */
+    public function createOrReplace($id, array $data)
```

Comment:
I'm not a fan of having the repository handle nested entity creation. I think we need to revisit how the controller interacts with the models.